### PR TITLE
improvements to the observabaility strategy

### PR DESCRIPTION
**Pull request by a-thaler**
**Description**

Changes proposed in this pull request:

- re-arranged tracing/pocs
- improved tracing concept
- improved strategy
- ...

**Related issue(s)**

### CODE REVIEW

File: concepts/observability-strategy/configurable-monitoring/04-monitoring-api.md [hunk 1]
```diff
@@ -7,89 +7,94 @@ apiVersion: telemetry.kyma-project.io/v1alpha1
...
   name: PrometheusRemoteWrite
 spec:
-  receiver: # singular, different receivers need different pipelines and exporterssss configs
-    type: application # application | system | custom, default is application
-    otlp: # maps to central tail pipeline, dealing with the actual application logs
-      namespaces: [] # generates the rule for the rewrite_tag assigned to every pipeline
-      excludeNamespaces: []
-      containers: []
-      excludeContainers: []
+  receiver: # singular, different receivers need different pipelines and exporters configs
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-monitoring/04-monitoring-api.md:10]: Not sure what you mean with "pipelines and exporters configs":

- different configs for pipeline and exporter

```suggestion
  receiver: # singular, different receivers need different pipeline and exporter configs
```

- or different pipelines, and different exporter configs

```suggestion
  receiver: # singular, different receivers need different pipelines and exporter configs
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 2]
```diff
@@ -2,9 +2,32 @@
 
 ## Requirements
 
+The agent's responsibility is to collect all span data in the cluster (Istio, Serverless, Eventing, custom) and ship them in a configurable way to targets. The agent is not meant to be used as backend system, so ideally it is even stateless.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:5]:
```suggestion
The agent's responsibility is to collect all span data in the cluster (Istio, Serverless, Eventing, custom) and ship them in a configurable way to targets. The agent is not meant to be used as backend system, so ideally, it is stateless.
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 3]
```diff
@@ -2,9 +2,32 @@
...
+
+- Collection
+  - Support OTLP push approach, ideally with service discovery. A lot of apps are based on the pull approach and should not be rewritten or operated with a sidecar only for transforming that approach.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:8]:
```suggestion
  - Support OTLP push approach, ideally with service discovery. Many apps are based on the pull approach and should not have to be rewritten or operated with a sidecar, only to transform that approach.
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 4]
```diff
@@ -2,9 +2,32 @@
...
+  - Support push-based model using OTLP.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:9]: Isn't this identical to line 8, just in fewer words?
Is line 8 really about push approach, or did you mean `pull` approach there (which would match the 2nd sentence better)

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 5]
```diff
@@ -2,9 +2,32 @@
...
+  - Scale-out must be possible dependent on incoming traffic
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:10]:
```suggestion
  - Scale-out must be possible, depending on incoming traffic
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 6]
```diff
@@ -2,9 +2,32 @@
...
+- Filter
+  - Support filtering traces/spans per pipeline based on the following criteria:
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:12]:
```suggestion
  - Support filtering traces or spans per pipeline, based on the following criteria:
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 7]
```diff
@@ -2,9 +2,32 @@
...
+     - Source by Namespace or Pod name and Kubernetes labels
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:13]: yay perfect capitalization!

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 8]
```diff
@@ -2,9 +2,32 @@
...
+  - Simple tail-based sampling based on percentage, potentially also by other criterias
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:14]:
```suggestion
  - Simple tail-based sampling based on percentage, potentially also by other criteria
```

"criteria" is already [plural](https://www.merriam-webster.com/dictionary/criteria). If you mean one, it's "criterion".

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 9]
```diff
@@ -2,9 +2,32 @@
...
+- Output
+  - Independent outputs (with buffer or retry management)
+  - OTLP support
 
 ## Alternatives
 
+### Jaeger Collector
+
+The Jaeger Collector is providing support for several input like OTLP, however, the outputs are related to backends support by Jaeger like Cassandra. It is not meant to be used to integrate other systems via OTLP.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:23]:
```suggestion
The Jaeger Collector provides support for several inputs like OTLP; however, the outputs are related to backends support by Jaeger, like Cassandra. It is not meant to be used to integrate other systems with OTLP.
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 10]
```diff
@@ -2,9 +2,32 @@
...
+
+### Grafana Agent
+
+License is problematic
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:27]:
```suggestion
License is problematic.
```

File: concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md [hunk 11]
```diff
@@ -2,9 +2,32 @@
...
+### OpenCensus Collector
+
+OpenCensus will be deprecated, OpenTelemetry should be used instead
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/02-tracing-collectors.md:30]:
```suggestion
OpenCensus will be deprecated. OpenTelemetry should be used instead.
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 12]
```diff
@@ -30,9 +30,13 @@ Running a Istio performance test with high load, and compare the following setti
 - 100% sampling to non-existing URL
 Comparing the resource consumption and throughput of the envoys, and checking other relevant Kubernetes components for suspicious effects like CoreDNS.
 
+Conclusions:
+- Client-based sampling should not be set to 100% by default, it must be a user controlled configuration
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:34]:
```suggestion
- Client-based sampling should not be set to 100% by default, it must be a user-controlled configuration
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 13]
```diff
@@ -30,9 +30,13 @@ Running a Istio performance test with high load, and compare the following setti
...
+- Tracing should be turned off if there is no endpoint available to push the data
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:35]:
```suggestion
- If there is no endpoint available to push the data, tracing should be deactivated
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 14]
```diff
@@ -30,9 +30,13 @@ Running a Istio performance test with high load, and compare the following setti
...
+
 ## Tail-based Sampling
 
-Opposed to head-based sampling, tail-based sampling makes the decision at the end of the entire flow, wenn all the trace data has been gathered. This kind of sampling decision is made at the collector level.
+Opposed to head-based sampling, tail-based sampling makes the decision at the end of the entire flow, when all the trace data has been gathered. This kind of sampling decision is made at the collector level.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:39]: oops my bad

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 15]
```diff
@@ -46,7 +50,7 @@ At the first glance, tail-based sampling seems to be a better solution than head
 
 Policy-based sampling processor configuration offers wide capabilities to configure sampling according to application needs, but this capability brings its own complexity. 
 
-At the time this document is written (Sept '21), tail-based sampling processor is still in a *beta* state and not fully tested.
+At the time this document is written (Sept '22), tail-based sampling processor is still in a *beta* state and not fully tested.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:53]: 😵‍💫

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 16]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
 ## How pipeline isolation can be achieved, is it feasible at all?
 
 The goal of the TracePipeline is to push trace data to multiple destinations using a different set of processors or sampling strategies. How can an isolation of these pipelines be achieved based on the otel-collector? If one destination is down, can the other destination be continued?
+
+## Modularization - 3 modules or one telemetry module
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:90]:
```suggestion
## Modularization - three modules or one telemetry module
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 17]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
...
+
+Benefits of one module
+- Shared caches for the controllers possible
+- less maintenance (3 images instead of one, think of security scanning, kube-builder updates)
+- shared packages possible without spending time in dependency tree maintenance
+
+Benefits of three modules
+- There are 3 independent domains sharing a lot of common things, it sounds more natural to model them individual
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:98]:
```suggestion
- There are 3 independent domains sharing a lot of common things, it sounds more natural to model them individually
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 18]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
...
+- Feature selection (on/off) is natively supported (no sub-attributes which are doing something similar)
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:99]:
```suggestion
- Feature selection (on/off) is natively supported (no sub-attributes with similar purposes)
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 19]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
...
+
+## Modularization - Re-use the deployer library of the module-manager
+For managing the otel-collector deployment, can the new library of the module-manager be used already? Is deletion supported? What artifact format should be used?
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:102]:
```suggestion
Can the new library of the Module Manager be used already to manage the Otel Collector deployment? Is deletion supported? What artifact format should be used?
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 20]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
...
+
+## Otel-Collector base setup
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:104]:
```suggestion
## Otel Collector base setup
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 21]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
...
+What processors and extensions the base setup needs to use, what are the configuration options?
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:105]:
```suggestion
Which processors and extensions does the base setup need? What are the configuration options?
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 22]
```diff
@@ -30,9 +30,13 @@ Running a Istio performance test with high load, and compare the following setti
...
+Conclusions:
+- Client-based sampling should not be set to 100% by default, it must be a user controlled configuration
+- Tracing should be turned off if there is no endpoint available to push the data
+
 ## Tail-based Sampling
 
-Opposed to head-based sampling, tail-based sampling makes the decision at the end of the entire flow, wenn all the trace data has been gathered. This kind of sampling decision is made at the collector level.
+Opposed to head-based sampling, tail-based sampling makes the decision at the end of the entire flow, when all the trace data has been gathered. This kind of sampling decision is made at the collector level.
 
 With tail-based sampling, it's possible to create advanced rules to filter out traces based on any **span** property, including their attributes, duration etc. With tail-based sampling, we can collect data like usually long operations and rare errors.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:41]:
```suggestion
With tail-based sampling, it's possible to create advanced rules to filter out traces based on any **span** property, including their attributes, duration etc. With tail-based sampling, we can collect data like unusually long operations and rare errors.
```

I think this was supposed to say "unusually", not quite sure.

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 23]
```diff
@@ -2,5 +2,32 @@
 
 ## Architecture Overview
 
+![b](./assets/tracing-future.drawio.svg)
 
+## Simple Auto-Scaling - Tail-based sampling
 
+An auto-scaling should be possible from the beginning. Hereby, a simple Deployment of the otel-collector will be the best option to achieve a scaling. Only relevant scaling criterias based on incoming traffic needs to be evaluated to scale up/down the replicaset. A single Deployment of the otel-collector requires that sampling decisions are made on the data which is constant and available across all spans of a trace, mainly the traceID. Any other sampling would require to batch trace data and do that batching sticky on dedicated instances. That would require an advanced architecture which could be switched to at a later point in time. A simple sampling approach is acceptable.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:9]:
```suggestion
Auto-scaling should be possible from the beginning. A simple Deployment of the Otel Collector is the best option to achieve scaling. We only have to evaluate the relevant scaling criteria based on incoming traffic to scale the ReplicaSet up and down. A single Deployment of the Otel Collector requires that sampling decisions are made on the data that is constant and available across all spans of a trace, mainly the traceID.
Any other sampling would require to batch trace data, and to do that batching sticky on dedicated instances. That would require an advanced architecture, which we can adopt later.
A simple sampling approach is acceptable for now.
```

I have no idea what this means: `do that batching sticky` ??

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 24]
```diff
@@ -2,5 +2,32 @@
...
+
+## Temporary OpenCensus support
+
+When introducing the configurable tracing, w3c-tracecontext should be used from the beginning. That requires to use the OpenCensus tracer in the istio configuration. That protocol needs to be supported in the beginning as a receiver till OTLP support is available.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:13]:
```suggestion
When introducing the configurable tracing, w3c-tracecontext should be used from the beginning. That requires to use the OpenCensus tracer in the Istio configuration. Till OTLP support is available, that protocol must be supported as a receiver.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 25]
```diff
@@ -2,5 +2,32 @@
...
+
+## Custom base image
+
+Not used plugins for the otel-collector should not be installed in the used image to reduce the attack vector. With that a custom bundling of the otel-collector image will be used
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:17]:
```suggestion
Unused plugins for the Otel Collector should not be installed in the used image to reduce the attack vector. With that, a custom bundling of the Otel Collector image will be used
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 26]
```diff
@@ -2,5 +2,32 @@
...
+
+## Operator manages the Otel-Collector Deployment
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:19]:
```suggestion
## Operator manages the Otel Collector Deployment
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 27]
```diff
@@ -2,5 +2,32 @@
...
+There should be no resource consumption if there is no pipeline defined. That will be achieved by managing the otel-collector fully by the telemetry-operator. So the operator will deploy and configure and potentially also delete the Deployment
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:20]:
```suggestion
There should be no resource consumption if there is no pipeline defined. To achieve this, the Telemetry Operator will fully manage the Otel Collector; meaning the operator will deploy, configure, and potentially also delete the Deployment.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 28]
```diff
@@ -2,5 +2,32 @@
...
+
+## Action Items
+
+Following work packages are identified and will be transformed into stories:
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:24]:
```suggestion
The following work packages are identified and will be transformed into stories:
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 29]
```diff
@@ -2,5 +2,32 @@
...
+- build a custom ote-collector image (with receiver for istio traces and tail-based sampling processor)
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:25]:
```suggestion
- Build a custom Otel Collector image with a receiver for Istio traces and tail-based sampling processor.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 30]
```diff
@@ -2,5 +2,32 @@
...
+- operator deploys otel-collector if a pipeline is defined
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:26]:
```suggestion
- If a pipeline is defined, the Telemetry Operator deploys Otel Collector.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 31]
```diff
@@ -2,5 +2,32 @@
...
+- operator removes deployment if last pipeline gets deleted
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:27]:
```suggestion
- When the last pipeline is deleted,  the Telemetry Operator removes the Deployment.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 32]
```diff
@@ -2,5 +2,32 @@
...
+- Have base settings in place for otel-collector deployment
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:28]:
```suggestion
- Define basic settings for the Otel Collector Deployment.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 33]
```diff
@@ -2,5 +2,32 @@
...
+- Configure otel-collector with tracePipeline settings
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:29]:
```suggestion
- Configure Otel Collector with tracePipeline settings.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 34]
```diff
@@ -2,5 +2,32 @@
...
+- Limit setup to a singular pipeline only (there can be only 0 or 1 pipeline to keep it simple)
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:30]:
```suggestion
- Limit the setup to a singular pipeline only. To keep it simple, there can be only 0 or 1 pipeline.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 35]
```diff
@@ -2,5 +2,32 @@
...
+- auto-scaling vs vertical scaling options
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:31]:
```suggestion
- Define auto-scaling vs vertical scaling options.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 36]
```diff
@@ -2,5 +2,32 @@
...
+- Expose relevant otel-collector metrics for troubleshooting
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:32]:
```suggestion
- Expose relevant Otel Collector metrics for troubleshooting.
```

File: concepts/observability-strategy/configurable-tracing/pocs/sampling-rate/README.md [hunk 37]
```diff
@@ -13,15 +13,19 @@ OpenTelemetry version 0.22.1 deployed on a Kyma cluster (version 2.6.0) with the
 - As exporter, only log/info and/or NOP exporter to keep impact of OpenTelemetry as low as possible.
 - No additional extensions
 
+See the [otel-collector](./files/otel-collector/) for details.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/pocs/sampling-rate/README.md:16]:
```suggestion
For details, see the [Otel Collector](./files/otel-collector/).
```

File: concepts/observability-strategy/configurable-tracing/pocs/sampling-rate/README.md [hunk 38]
```diff
@@ -13,15 +13,19 @@ OpenTelemetry version 0.22.1 deployed on a Kyma cluster (version 2.6.0) with the
...
 As sampling app, three serverless Functions are deployed with Istio sidecar injected. To simulate a trace fully sampled with an external call, the Functions call each other in the following chain:
 **Extern Call -> FunctionA -> FunctionB -> FunctionC**
 
 All Functions are deployed with NodeJs version 16.
 
+See the [application folder](./files/application/) for details.
+
 ### Call Simulator
-As call simulator, Gatling version 3.8.3 is used.
+As call simulator, Gatling version 3.8.3 is used, see the [Gatling test](./files/call-simulator/TracingSimulation.java)
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/pocs/sampling-rate/README.md:28]:
```suggestion
As call simulator, Gatling version 3.8.3 is used; see the [Gatling test](./files/call-simulator/TracingSimulation.java).
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 39]
```diff
@@ -21,7 +21,7 @@ To support sampling centrally with the pipeline in the collector config, head-ba
 
 For that, it is crucial to understand the impact of a permanently enabled sampling strategy on the used infrastructure, mainly Istio and Kubernetes.
 
-The [Istio trace sampling rate analysis](https://github.com/kyma-project/kyma/issues/15304) investigation does that in the following way:
+The [Istio trace sampling rate analysis](./pocs/sampling-rate/README.md) investigation does that in the following way:
 
 Running a Istio performance test with high load, and compare the following settings:
 - 1% sampling to otel-collector with noop exporter
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:27]:
```suggestion
- 1% sampling to Otel Collector with noop exporter
```

File: concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md [hunk 40]
```diff
@@ -82,3 +86,20 @@ Istio can be configured to use the w3c-tracecontext for trace propagation alread
 ## How pipeline isolation can be achieved, is it feasible at all?
 
 The goal of the TracePipeline is to push trace data to multiple destinations using a different set of processors or sampling strategies. How can an isolation of these pipelines be achieved based on the otel-collector? If one destination is down, can the other destination be continued?
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/03-tracing-pocs.md:88]:
```suggestion
The goal of the TracePipeline is to push trace data to multiple destinations using a different set of processors or sampling strategies. How can an isolation of these pipelines be achieved based on the Otel Collector? If one destination is down, can the other destination be continued?
```

**Review by NHingerl — COMMENTED** [review]
for consistency

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 41]
```diff
@@ -2,5 +2,34 @@
 
 ## Architecture Overview
 
+![b](./assets/tracing-future.drawio.svg)
 
+## Simple Auto-Scaling - Tail-based sampling
 
+Auto-scaling should be possible from the beginning. A simple Deployment of the Otel Collector is the best option to achieve scaling. We only have to evaluate the relevant scaling criteria based on incoming traffic to scale the ReplicaSet up and down. A single Deployment of the Otel Collector requires that sampling decisions are made on the data that is constant and available across all spans of a trace, mainly the traceID.
+Any other sampling would require to batch trace data, and to do that batching sticky on dedicated instances. That would require an advanced architecture, which we can adopt later.
```
Comment (a-thaler) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:10]:
```suggestion
More advanced sampling like respecting some error status would require to batch all spans for a trace at one instance in order to make a decision. Having a scalable setup would require to introduce another StatefulSet of the Otel Collector and forward the spans for a specific trace always to the same instance of the StatefulSet (sticky). That advanced architecture, can be adopted at a later time.
```

File: concepts/observability-strategy/configurable-tracing/05-tracing-concept.md [hunk 42]
```diff
@@ -7,7 +7,7 @@
 ## Simple Auto-Scaling - Tail-based sampling
 
 Auto-scaling should be possible from the beginning. A simple Deployment of the Otel Collector is the best option to achieve scaling. We only have to evaluate the relevant scaling criteria based on incoming traffic to scale the ReplicaSet up and down. A single Deployment of the Otel Collector requires that sampling decisions are made on the data that is constant and available across all spans of a trace, mainly the traceID.
-Any other sampling would require to batch trace data, and to do that batching sticky on dedicated instances. That would require an advanced architecture, which we can adopt later.
+More advanced sampling like respecting some error status would require to batch all spans for a trace at one instance in order to make a decision. Having a scalable setup would require to introduce another StatefulSet of the Otel Collector and forward the spans for a specific trace always to the same instance of the StatefulSet (sticky). That advanced architecture, can be adopted at a later time.
```
Comment (NHingerl) [inline concepts/observability-strategy/configurable-tracing/05-tracing-concept.md:10]:
```suggestion
More advanced sampling like respecting some error status would require to batch all spans for a trace at one instance in order to make a decision. Having a scalable setup would require to introduce another StatefulSet of the Otel Collector and forward the spans for a specific trace always to the same instance of the StatefulSet (sticky). That advanced architecture can be adopted at a later time.
```

**Review by NHingerl — APPROVED** [review]

**Pull request merged**